### PR TITLE
Add status for unknown user's filter ID

### PR DESCRIPTION
**Pull request by maxidor**
This PR fixes a spec omission about the specific status code returned when an unknown user's filter ID is requested.

Signed-off-by: Max Dor @ Kamax.io (no email to avoid spam)

### CODE REVIEW

File: api/client-server/filter.yaml [hunk 1]
```diff
@@ -147,5 +147,7 @@ paths:
             type: object
             allOf:
               - $ref: "definitions/sync_filter.yaml"
+        400:
```
Comment (richvdh) [inline api/client-server/filter.yaml]: shouldn't this be a 404? (Even if synapse is returning 400 here)

Comment (maxidor) [reply api/client-server/filter.yaml]: I don't know what should be the status code.This PR is only to document existing behavior so the community can benefit from it.

Comment (richvdh) [reply api/client-server/filter.yaml]: Let me put it more strongly then. "I think this should be a 404 and synapse should be fixed".

Comment (maxidor) [reply api/client-server/filter.yaml]: Set to `404` following your feedback.

**Comment by richvdh** [comment]
:+1:

**Pull request merged**